src/SuperVault.sol

```diff
@@ -34,74 +39,108 @@ contract SuperVault is BaseStrategy, ISuperVault {
     //                       MODIFIERS                          //
     //////////////////////////////////////////////////////////////
 
     /// @notice Ensures that only the Super Vaults Strategist can call the function
     modifier onlySuperVaultsStrategist() {
-        if (_getAddress(keccak256("SUPER_VAULTS_STRATEGIST")) != msg.sender) {
+        if (strategist != msg.sender) {
             revert NOT_SUPER_VAULTS_STRATEGIST();
         }
         _;
     }
 
+    /// @notice Ensures that only the Vault Manager can call the function
+    modifier onlyVaultManager() {
+        if (vaultManager != msg.sender) {
+            revert NOT_VAULT_MANAGER();
+        }
+        _;
+    }
+
     //////////////////////////////////////////////////////////////
     //                       CONSTRUCTOR                        //
     //////////////////////////////////////////////////////////////
 
     /// @param superRegistry_ Address of the SuperRegistry contract
     /// @param asset_ Address of the asset token
-    /// @param refundsReceiver_ Address to receive refunds
     /// @param name_ Name of the strategy
     /// @param depositLimit_ Maximum deposit limit
     /// @param superformIds_ Array of Superform IDs
     /// @param startingWeights_ Array of starting weights for each Superform
     constructor(
         address superRegistry_,
         address asset_,
-        address refundsReceiver_,
+        address strategist_,
+        address vaultManager_,
         string memory name_,
         uint256 depositLimit_,
         uint256[] memory superformIds_,
         uint256[] memory startingWeights_
     )
         BaseStrategy(asset_, name_)
     {
-        if (superRegistry_ == address(0) || refundsReceiver_ == address(0)) {
-            revert ZERO_ADDRESS();
+        numberOfSuperforms = superformIds_.length;
+
+        if (numberOfSuperforms == 0) {
+            revert ZERO_SUPERFORMS();
         }
 
-        if (block.chainid > type(uint64).max) {
-            revert BLOCK_CHAIN_ID_OUT_OF_BOUNDS();
+        if (numberOfSuperforms != startingWeights_.length) {
+            revert ARRAY_LENGTH_MISMATCH();
+        }
+
+        if (superRegistry_ == address(0) || strategist_ == address(0) || vaultManager_ == address(0)) {
+            revert ZERO_ADDRESS();
         }
 
         CHAIN_ID = uint64(block.chainid);
 
         superRegistry = ISuperRegistry(superRegistry_);
-        refundReceiver = refundsReceiver_;
+        superformFactory = ISuperformFactory(superRegistry.getAddress(keccak256("SUPERFORM_FACTORY")));
 
-        uint256 numberOfSuperforms = superformIds_.length;
-        if (numberOfSuperforms != startingWeights_.length) {
-            revert ARRAY_LENGTH_MISMATCH();
-        }
+        CHAIN_ID = uint64(block.chainid);
 
-        ISuperformFactory factory = ISuperformFactory(_getAddress(keccak256("SUPERFORM_FACTORY")));
+        if (CHAIN_ID > type(uint64).max) {
```

Comment:
this check doesn't work, because `CHAIN_ID = uint64(block.chainid);` has already truncated the value

solution:
```solidity
if (block.chainid > type(uint64).max) revert;
CHAIN_ID = uint64(block.chainid);
```

Reply:
https://github.com/superform-xyz/SuperVaults/pull/30/